jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Jstack.java

```diff
@@ -0,0 +1,156 @@
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package openj9.tools.attach.diagnostics.tools;
+
+import static java.lang.System.err;
+import static java.lang.System.out;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.ibm.tools.attach.target.AttachHandler;
+import com.ibm.tools.attach.target.IPC;
+
+import openj9.tools.attach.diagnostics.attacher.AttacherDiagnosticsProvider;
+import openj9.tools.attach.diagnostics.base.DiagnosticProperties;
+import openj9.tools.attach.diagnostics.base.DiagnosticsInfo;
+
+/**
+ * JStack 
+ * A tool for listing thread information about another Java process
+ *
+ */
+public class Jstack {
+
+	private static List<String> vmids;
+	private static boolean printProperties;
+	private static boolean printSynchronizers;
+	/**
+	 * Print a list of Java processes and information about them.
+	 * @param args Arguments to the application
+	 */
+	public static void main(String[] args) {
+		int rc = 0;
+		if (!parseArguments(args)) {
+			System.exit(1);
+		}
+		AttacherDiagnosticsProvider diagProvider = new AttacherDiagnosticsProvider();
+
+
+		String myId = AttachHandler.getVmId();
+		out.println(LocalDateTime.now().toString());
+		for (String vmid: vmids) {
+			if (vmid.equals(myId)) {
+				continue;
+			}
+			/* if the ID looks like a process ID, check if it is running */
+			if (vmid.matches("\\d+")) { //$NON-NLS-1$
+				long pid = Long.parseLong(vmid);
+				if (!IPC.processExists(pid)) {
+					continue;
+				}
+			}
+			try {
+				diagProvider.attach(vmid);
+				DiagnosticsInfo groupInfo = diagProvider.getThreadGroupInfo(printSynchronizers);
+				out.printf("Virtual machine: %s JVM information %s%n", vmid, groupInfo.getJavaInfo()); //$NON-NLS-1$
+				out.println(groupInfo.toString());
+				if (printProperties) {
+					out.println("System properties:"); //$NON-NLS-1$
+					out.println(diagProvider.getSystemProperties());
+					out.println("Agent properties:"); //$NON-NLS-1$
+					out.println(diagProvider.getAgentProperties());
+				}
+			} catch (Exception e) {
+				System.err.printf("Error getting data from %s", vmid); //$NON-NLS-1$
+				final String msg = e.getMessage();
+				if (null == msg) {
+					err.printf("%n"); //$NON-NLS-1$
+				} else {
+					err.printf(": %s%n", msg); //$NON-NLS-1$
+				}
+				if (DiagnosticProperties.isDebug) {
+					e.printStackTrace();
+				}
+			} finally {
+				try {
+					diagProvider.detach();
+				} catch (IOException e) {
+					/* ignore */
+				}
+			}
+		}
+		System.exit(rc);
```

Comment:
This doesn't appear to be useful: I don't see any changes after `int rc = 0;`.

Reply:
True.  Removed it and the variable.